File: remove_conflicts.py
#!/usr/bin/env python3
import os
import re
from pathlib import Path

def remove_merge_conflicts(directory, extensions):
    """Remove merge conflict markers, keeping the HEAD version"""
    fixed_count = 0
    pattern = r'<<<<<<< HEAD\n([\s\S]*?)\n=======\n[\s\S]*?\n>>>>>>> [^\n]+\n'

    for root, dirs, files in os.walk(directory):
        # Skip node_modules and dist
        if 'node_modules' in root or 'dist' in root or '.git' in root:
            continue

        for file in files:
            ext = Path(file).suffix
            if ext in extensions:
                filepath = os.path.join(root, file)
                try:
                    with open(filepath, 'r', encoding='utf-8') as f:
                        content = f.read()

                    original_len = len(content)
                    # Remove conflict markers, keep HEAD version
                    new_content = re.sub(pattern, r'\1', content)

                    if len(new_content) != original_len:
                        with open(filepath, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                        fixed_count += 1
                        print(f"✓ Fixed: {filepath}")
                except Exception as e:
                    print(f"✗ Error processing {filepath}: {str(e)}")

    print(f"\n✅ Removed merge conflict markers from {fixed_count} files")

extensions = ['.ts', '.tsx', '.js', '.jsx', '.css', '.json', '.html', '.md', '.yml', '.yaml']
remove_merge_conflicts('.', extensions)

File: remove_conflicts.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

function removeConflicts(dir, extensions) {
  const files = fs.readdirSync(dir, { recursive: true, encoding: 'utf-8' });
  let fixedCount = 0;

  for (const file of files) {
    // Skip directories
    if (file.includes('node_modules') || file.includes('dist') || file.includes('.git')) {
      continue;
    }

    const ext = path.extname(file);
    if (extensions.includes(ext)) {
      const fullPath = path.join(dir, file);
      const stat = fs.statSync(fullPath);

      if (stat.isFile()) {
        try {
          let content = fs.readFileSync(fullPath, 'utf-8');
          const originalLength = content.length;

          // Remove merge conflict markers, keeping HEAD version
          content = content.replace(/<<<<<<< HEAD\n([\s\S]*?)\n=======\n[\s\S]*?\n>>>>>>> [^\n]+\n/g, '$1');

          if (content.length !== originalLength) {
            fs.writeFileSync(fullPath, content, 'utf-8');
            fixedCount++;
            console.log(`✓ Fixed: ${file}`);
          }
        } catch (err) {
          console.error(`✗ Error processing ${file}:`, err.message);
        }
      }
    }
  }

  console.log(`\n✅ Removed merge conflict markers from ${fixedCount} files`);
}

const extensions = ['.ts', '.tsx', '.js', '.jsx', '.css', '.json', '.html', '.md', '.yml', '.yaml'];
removeConflicts('.', extensions);
